packages/flutter_tools/lib/src/build_runner/resident_web_runner.dart

```diff
@@ -530,6 +532,13 @@ class _ExperimentalResidentWebRunner extends ResidentWebRunner {
       });
       _wipConnection = await chromeTab.connect();
     }
+    final WebDevFS webDevFS = device.devFS as WebDevFS;
+    final Dwds dwds = webDevFS.dwds;
+    dwds.connectedApps.listen((AppConnection appConnection) async {
+      appConnection.runMain();
+      _debugConnection = await dwds.debugConnection(appConnection);
```

Comment:
To do things with the compile expression support, you'd want to grab the vmservice here and register a new method that forwards to the underling resident compiler